protocols/mdns/src/behaviour.rs

```diff
@@ -347,8 +347,10 @@ where
         }
 
         if !discovered.is_empty() {
-            let event = Event::Discovered(discovered);
-            return Poll::Ready(ToSwarm::GenerateEvent(event));
+            return Poll::Ready(ToSwarm::NewExternalAddrOfPeer {
```

Comment:
[question]

Sorry if this is a silly question, I'm new to both rust and the libp2p code base.

1) This seems like a breaking change for any code base that currently relies on `Event::Discovered`

Is my understanding of rust breaking changes right here?

Would it be worth preserving `Event::Discovered` here for backwards compatibility?

2) This only emits the first discovered peer

Reading lines 333-345, it looks like the Poll event will keep going until it drains query_response_receiver, discovering multiple peers?

This seems like we would want to return a `ToSwarm::NewExternalAddrOfPeer` for every peer we discover on line 345.

Should this move up to line 345 and replace the discovered vec?

3)

I acknowledge that, as written, point 2 might be at odds with point 1 :sweat_smile:.

To have our cake, and eat it too, could 333-347 be moved out of `fn poll` and instead feed two receivers, one for `Event::Discovered` once all `ToSwarm::NewExternalAddrOfPeer`s have been flushed?

Reply:
yeah exactly, thanks @retrohacker!
You can see the [upnp example](https://github.com/libp2p/rust-libp2p/blob/master/protocols/upnp/src/behaviour.rs#L422-L435) on how to return both